models/dns.go

```diff
@@ -60,74 +61,134 @@ type DNSProviderConfig struct {
 //    This is the FQDN version of Name.
 //    It should never have a trailiing ".".
 type RecordConfig struct {
-	Type     string            `json:"type"`
-	Name     string            `json:"name"`   // The short name. See below.
-	Target   string            `json:"target"` // If a name, must end with "."
-	TTL      uint32            `json:"ttl,omitempty"`
-	Metadata map[string]string `json:"meta,omitempty"`
-	NameFQDN string            `json:"-"` // Must end with ".$origin". See below.
-	Priority uint16            `json:"priority,omitempty"`
+	Type        string            `json:"type"`
+	Name        string            `json:"name"`   // The short name. See below.
+	Target      string            `json:"target"` // If a name, must end with "."
+	TTL         uint32            `json:"ttl,omitempty"`
+	Metadata    map[string]string `json:"meta,omitempty"`
+	NameFQDN    string            `json:"-"`                  // Must end with ".$origin". See below.
+	Priority    uint16            `json:"priority,omitempty"` // MX
+	SrvPriority uint16            `json:"SrvPriority,omitempty"`
+	SrvWeight   uint16            `json:"SrvWeight,omitempty"`
+	SrvPort     uint16            `json:"SrvPort,omitempty"`
 
 	Original interface{} `json:"-"` // Store pointer to provider-specific record object. Used in diffing.
 }
 
-func (r *RecordConfig) String() string {
-	content := fmt.Sprintf("%s %s %s %d", r.Type, r.NameFQDN, r.Target, r.TTL)
-	if r.Type == "MX" {
-		content += fmt.Sprintf(" priority=%d", r.Priority)
+func (r *RecordConfig) String() (content string) {
+	switch r.Type {
+	case "A", "AAAA", "PTR":
+		content = fmt.Sprintf("%s %s %s %d", r.Type, r.NameFQDN, r.Target, r.TTL)
+	case "MX":
+		content = fmt.Sprintf(" priority=%d", r.Priority)
+	case "SOA":
+		content = fmt.Sprintf("%s %s %s %d", r.Type, r.Name, r.Target, r.TTL)
+	default:
+		panic(fmt.Sprintf("rc.String rtype %v unimplemented", r.Type))
 	}
 	for k, v := range r.Metadata {
 		content += fmt.Sprintf(" %s=%s", k, v)
 	}
 	return content
 }
 
-/// Convert RecordConfig -> dns.RR.
-func (r *RecordConfig) ToRR() dns.RR {
+// Content combines Target and other fields into one string.
+func (r *RecordConfig) Content() string {
 
-	// Note: The label is a FQDN ending in a ".".  It will not put "@" in the Name field.
+	// If this is a pseudo record, just return the target.
+	if _, ok := dns.StringToType[r.Type]; !ok {
+		return r.Target
+	}
 
-	// NB(tlim): An alternative way to do this would be
-	// to create the rr via: rr := TypeToRR[x]()
-	// then set the parameters. A benchmark may find that
-	// faster. This was faster to implement.
+	// We cheat by converting to a dns.RR and use the String() function.
+	// Sadly that function always includes a header, which we must strip out.
+	// TODO(tlim): Request the dns project add a function that returns
+	// the string without the header.
```

Comment:
I'm not sure I love relying on the dns package for this, especially with the number of caveats we add. Maybe just implement our own switch?